Slim/Schema/Composer.pm

```diff
@@ -0,0 +1,56 @@
+package Slim::Schema::Composer;
+
+
+use strict;
+use base 'Slim::Schema::DBI';
+
+use Scalar::Util qw(blessed);
+
+use Slim::Schema::ResultSet::Composer;
+
+use Slim::Utils::Log;
+use Slim::Utils::Misc;
+
+my %contributorToRoleMap = (
+	'ARTIST'      => 1,
+	'COMPOSER'    => 2,
+	'CONDUCTOR'   => 3,
+	'BAND'        => 4,
+	'ALBUMARTIST' => 5,
+	'TRACKARTIST' => 6,
+);
+
+my @contributorRoles = sort keys %contributorToRoleMap;
+my @contributorRoleIds = values %contributorToRoleMap;
+my $totalContributorRoles = scalar @contributorRoles;
+
+my %roleToContributorMap = reverse %contributorToRoleMap;
+
+{
+	my $class = __PACKAGE__;
+
+	$class->table('contributors');
+
+	$class->add_columns(qw(
+		id
+		name
+		namesort
+		musicmagic_mixable
+		namesearch
+		musicbrainz_id
+		extid
+	));
+
+	$class->set_primary_key('id');
+	$class->add_unique_constraint('namesearch' => [qw/namesearch/]);
+
+	if ($] > 5.007) {
+		$class->utf8_columns(qw/name namesort/);
+	}
```

Comment:
OK. This is just copy/pasted from Contributor.pm, because I need a separate class for composer to make the DBIx stuff work. I'll remove the condition.